microprofile/design-doc-template.adoc

```diff
@@ -0,0 +1,178 @@
+---
```

Comment:
@kabir Could you rename the file `design-doc-template` before making the PR ready for review?